core/src/main/java/com/linecorp/armeria/server/file/ClassPathHttpFile.java

```diff
@@ -54,14 +57,21 @@ protected String pathOrUri() {
     }
 
     @Override
-    public HttpFileAttributes readAttributes() throws IOException {
-        if (attrs == null) {
-            final URLConnection conn = url.openConnection();
-            final long length = conn.getContentLengthLong();
-            final long lastModifiedMillis = conn.getLastModified();
-            attrs = new HttpFileAttributes(length, lastModifiedMillis);
+    public CompletableFuture<HttpFileAttributes> readAttributes(Executor fileReadExecutor) {
+        if (attrsFuture != null) {
+            return attrsFuture;
         }
-        return attrs;
+
+        return attrsFuture = CompletableFuture.supplyAsync(() -> {
+            try {
+                final URLConnection conn = url.openConnection();
+                final long length = conn.getContentLengthLong();
+                final long lastModifiedMillis = conn.getLastModified();
+                return new HttpFileAttributes(length, lastModifiedMillis);
+            } catch (IOException e) {
+                return Exceptions.throwUnsafely(e);
+            }
+        });
```

Comment:
Shouldn't this be `}, fileReadExecutor);`?
And rnn for `fileReadExecutor`.

Reply:
😱